### ISSUE TITLE
Improve training to pass semantic evaluation metrics

### ISSUE THREAD

**Issue by francisbrero** [francisbrero/llm-fine-tuning#9]
## Problem

Semantic evaluation reveals the model has poor GTM context understanding:
- **Tool Selection Accuracy: 20%** (Target: 85%+) - Would make wrong tool calls 80% of the time
- **Critical Field Accuracy: 26.7%** (Target: 80%+) - Fails to infer critical fields correctly
- **Overall Semantic Score: 22.2%** (Target: 75%+) - Not production-ready

The exact field matching eval (41% accuracy) was misleading - it credited partial matches that don't translate to correct downstream tool selection.

## Root Cause

The current training data teaches the model to:
✅ Produce valid JSON with correct field names (100% JSON validity)
❌ Understand implicit GTM context and semantics

### Specific Failures

1. **"Show me SQLs from the webinar"** - Complete failure (0%)
   - Doesn't understand SQLs = Sales Qualified Leads
   - Likely confuses with database SQL queries

2. **"Help me prepare for my QBR"** - Wrong tool selection
   - Should understand QBR = review existing accounts
   - Fails to infer expansion/renewal motion

3. **"Who's ghosting us?"** - Wrong tool selection
   - Should understand "ghosting" = accounts going silent = churn risk
   - Fails slang interpretation

4. **"Net new ARR this quarter"** - Wrong tool selection
   - Should infer outbound motion on net_new accounts
   - Fails to connect "net new" with prospecting tools

**Only 1 out of 5 examples succeeded:** "Which accounts haven't been touched in 30 days?" correctly inferred churn risk.

## Solution

Enhance training data to focus on **GTM context understanding**, not just syntax:

### 1. Add Context-Rich Training Examples

For each training example, include reasoning that explains the GTM context:

```json
{
  "gtm_prompt": "Show me my best accounts",
  "reasoning": "- 'my accounts' implies EXISTING accounts (not prospects)\n- 'best' suggests looking for expansion opportunities\n- This is an account_discovery intent with expansion motion\n- User is likely a sales rep or account manager",
  "intent_ir": {
    "intent_type": "account_discovery",
    "motion": "expansion",
    "account_scope": "existing",
    ...
  }
}
```

### 2. Add Slang/Acronym Examples

The model fails on GTM jargon. Add more examples:
- "SQLs" = Sales Qualified Leads (not database queries)
- "ghosting" = going silent = churn risk
- "QBR" = Quarterly Business Review = existing account review
- "whitespace" = expansion opportunities
- "going dark" = churn risk signal

### 3. Add Tool Selection Validation

For each example, document which tool categories should be selected:

```json
{
  "gtm_prompt": "Which accounts are at risk?",
  "tool_categories_correct": ["risk_scoring", "engagement_analysis", "churn_prediction"],
  "tool_categories_wrong": ["prospecting", "lead_gen", "outbound_tools"],
  "intent_ir": {...}
}
```

### 4. Add Negative Examples

Examples where clarification is needed:
- Ambiguous prompts
- Missing critical context
- Multiple valid interpretations

### 5. Increase Training Examples

Current: ~1,278 examples
Target: 3,000+ examples with diverse:
- Roles (sales_rep, cs, sdr, manager, revops)
- Motions (outbound, inbound, expansion, churn_prevention)
- Account scopes (net_new, existing, churned)
- Slang and jargon variations

## Validation

Use semantic evaluation to track progress:

```bash
# Before training
python ccp_cli.py eval-semantic --model base

# After each training iteration
python ccp_cli.py eval-semantic --model finetuned --compare-with-exact

# Target metrics
- Critical Field Accuracy: 80%+
- Tool Selection Accuracy: 85%+
- Overall Semantic Score: 75%+
```

## Success Criteria

- [ ] Tool Selection Accuracy > 85% (currently 20%)
- [ ] Critical Field Accuracy > 80% (currently 26.7%)
- [ ] Overall Semantic Score > 75% (currently 22.2%)
- [ ] Zero catastrophic failures (0% scores)
- [ ] Handles slang/acronyms correctly (SQLs, ghosting, QBR)

## Implementation Steps

1. [ ] Audit current training data for context richness
2. [ ] Add reasoning field to all training examples
3. [ ] Create 100+ slang/acronym examples
4. [ ] Add tool selection annotations
5. [ ] Generate 2,000 more diverse examples
6. [ ] Retrain with enhanced dataset
7. [ ] Run semantic eval and iterate
8. [ ] Document new training data format in gtm_domain_knowledge.md

## References

- Semantic evaluation design: `docs/semantic_eval_design.md`
- Current results: `eval_results/eval_semantic_finetuned_20260111_111417.json`
- Tool category mappings: `ccp/eval/semantic_scoring.py`

## Related Commands

```bash
# Run semantic evaluation
python ccp_cli.py eval-semantic --quick

# Compare with exact matching
python ccp_cli.py eval-semantic --quick --compare-with-exact

# After retraining
python ccp_cli.py eval-semantic --model both
```
